isaric/parsers/ecmo_subject.toml

```diff
@@ -0,0 +1,229 @@
+  values = { 1 = true, 0 = false }
+
+[subject.has_chronic_cardiac_disease] # SK to check here!
+  description = "Chronic cardiac disease, including congenital heart disease (not hypertension)"
+  combinedType = "firstNonNull"
+  fields = [
+    { field = "chroniccard_mhyn", ref = "Y/N/NK", description = "Chronic cardiac disease" },
+    { field = "c_chd", values = { 1 = true, 0 = false }, description = "Congenital heart disease" },
+    { field = "cardiomyopathy_ceterm", ref = "Y/N/NK", description = "Cardiomyopathy" },
+    { field = "c_pcmyo", values = { 1 = true, 0 = false }, description = "Pre-existing Cardiomyopathy" },
+    { field = "c_diagtak", values = { 1 = true, 0 = false }, description = "DIAGNOSIS OF TAKOTSUBO CARDIOMYOPATHY" }
+  ]
+
+[subject.has_chronic_kidney_disease]
+  field = "renal_mhyn"
+  ref = "Y/N/NK"
+  description = "Chronic kidney disease"
+
+[subject.has_diabetes] # NEED TO CHECK THIS!
+  combinedType = "firstNonNull"
+  fields = [
+    { field = "diabetes_spain", description = "Did the patient have diabetes?" },
+    { field = "diabetiscomp_mhyn", ref = "Y/N/NK", description = "Diabetes with complications" },
+    { field = "diabetes_mhyn", ref = "Y/N/NK", description = "Diabetes without complications" },
+    { field = "diabetes_mhyn_2", values = {1 = true, 2 = false, 4 = true, 5 = true} } # based on diabetes type
+  ]
+
+[subject.diabetes_type]
+  field = "diabetes_mhyn_2"
+  description = "Diabetes and Type"
+  values = { 1 = "type-1", 4 = "type-2", 5 = "gestational" }
+
+[subject.has_liver_disease]
+  combinedType = "any"
+  fields = [
+    { field = "modliver_mhyn", description = "Moderate liver disease", values = { 1 = true, 0 = false } },
+    { field = "mildliv_mhyn", description = "Mild liver disease", values = { 1 = true, 0 = false } },
+  ]
+
+[subject.has_hiv_art]
+  field = "aidshiv_mhyn_2"
+  description = "patient is on ART (antiretroviral therapy)"
+  values = { 1 = true, 2 = false, 4 = false }
+
+[subject.has_immunosuppression_therapy_treatment]
+  field = "pre_nonoralsteroid_cmoccur"
+  ref = "Y/N/NK"
+  description = "Other immunosuppressant agents (not oral steroids)"
+
+[subject.has_solid_organ_transplant]
+  description = "Solid organ transplant recipients:"
+  combinedType = "firstNonNull"
+  fields = [
+    { field = "f_lungt_yn", values = { 1 = true, 0 = false }, description = "Did this patient receive a lung transplant due to COVID-19 associated pulmonary complications?" },
+    { field = "c_ptransplant", values = { 1 = true, 0 = false }, description = "Previous Cardiac transplant" }
+  ]
+
+[subject.date_death]
+  combinedType = "firstNonNull"
+  fields = [
+    { field = "f_datedeath", description = "Date of death", if = { dsterm = 4 } },
+    { field = "dsstdtc", description = "Outcome date", if = { dsstdtcyn = 1 } }
+  ]
+  
+[subject.icu_admitted]
+  field = "icu_hoterm"	
+  description = "Admitted to ICU?"
+  apply = { function = "isNotNull" } # WHAT DOES THIS DO?
```

Comment:
This uses the [isNotNull function](https://github.com/globaldothealth/adtl/blob/9904403d94edb8284016a7c4fbd3189f1dd66c28/adtl/transformations.py#L7) defined in adtl.transformations, so icu_admitted is true if icu_hoterm is not null.

Reply:
Understood.